### PR TITLE
fix: first screen view name not exists in rare cases

### PR DESCRIPTION
**Pull request by zhu-xiaowei**
## Description
1. fix first screen view name not exists in rare cases

This issue discovered by [e2e integration test](https://github.com/awslabs/clickstream-swift/actions/runs/7758708766/job/21161994170), and the integration test case will also cover this rare cases.

## General Checklist

- [x] Added new tests to cover change, if needed
- [x] Build succeeds using Swift Package Manager
- [x] All unit tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Sources/Clickstream/Dependency/Clickstream/AutoRecord/AutoRecordEventClient.swift [hunk 1]
```diff
@@ -64,12 +64,11 @@ class AutoRecordEventClient {
         if lastEngageTime > 0 {
             event.addAttribute(lastEngageTime, forKey: Event.ReservedAttribute.ENGAGEMENT_TIMESTAMP)
         }
-        recordEvent(event)
-
-        isEntrances = false
         lastScreenName = screenName
         lastScreenPath = screenPath
         lastScreenUniqueId = screenUniqueId
+        recordEvent(event)
```
Comment (zxkane) [inline Sources/Clickstream/Dependency/Clickstream/AutoRecord/AutoRecordEventClient.swift:70]: add test case for this change

Comment (zhu-xiaowei) [reply Sources/Clickstream/Dependency/Clickstream/AutoRecord/AutoRecordEventClient.swift:70]: this logic is covered by this [test case](https://github.com/awslabs/clickstream-swift/blob/main/Tests/ClickstreamTests/Clickstream/AutoRecordEventClientTest.swift#L91-L98)

Comment (zxkane) [reply Sources/Clickstream/Dependency/Clickstream/AutoRecord/AutoRecordEventClient.swift:70]: Why the test case was passed without this fix? Is it an intermitent issue?

Comment (zhu-xiaowei) [reply Sources/Clickstream/Dependency/Clickstream/AutoRecord/AutoRecordEventClient.swift:70]: Yes, the probability of this problem occurring is extremely low, and it is difficult to detect through unit testing. Only by conducting thousands of app startup tests on a real machine can it be discovered. So this issue only found by device farm integration test.
This modification ensures that the screen view name is always included.

**Pull request closed without merging**